lib/service/storage/elasticsearch.js

```diff
@@ -985,6 +985,66 @@ class ElasticSearch extends Service {
       throw this._esWrapper.formatESError(error);
     }
   }
+
+  /**
+  * Updates all documents matching the provided filters
+  *
+  * @param {String} index - Index name
+  * @param {String} collection - Collection name
+  * @param {Object} query - Query to match documents
+  * @param {Object} changes - Changes wanted on documents
+  * @param {Object} options - refresh (undefined)
+  *
+  * @returns {Promise.<{ successes: [_id, _source, _status], errors: [ document, status, reason ] }>}
+  */
+  async bulkUpdateByQuery (
+    index,
+    collection,
+    query,
+    changes,
+    { refresh = 'false' } = {})
+  {
+    const script = {
+      params: {},
+      source: ''
+    };
+
+    for (const [key, value] of Object.entries(changes)) {
+      script.source += `ctx._source.${key} = params['${key}'];`;
```

Comment:
Does this work for nested keys ?

Reply:
Maybe a deep flatten is required here? :thinking: